src/header.tsx

```diff
@@ -175,81 +187,95 @@ export const HeaderLayout: React.FC<HeaderLayoutProps> = ({
+                            ? 0
+                            : {
+                                  mobile: 24,
+                                  desktop: isBleedActivated && !sideBySideExtraOnDesktop ? 32 : 48,
+                              }
+                    }
+                >
+                    {sideBySideExtraOnDesktop ? (
                         <GridLayout
                             template="6+6"
-                            left={
-                                <Stack space={32}>
-                                    {breadcrumbs}
-                                    {header}
-                                </Stack>
+                            left={mainContent}
+                            right={
+                                <div className={isBleedActivated ? styles.hideOnTabletOrSmaller : ''}>
+                                    <Box paddingTop={{mobile: header ? 24 : 0, desktop: 0}}>{extra}</Box>
+                                </div>
                             }
-                            right={extra}
                         />
-                    </Box>
-                ) : (
-                    <Box paddingTop={breadcrumbs ? 16 : 48} paddingBottom={isBleedActivated ? 32 : 48}>
-                        <Stack space={isTabletOrSmaller ? 24 : 32}>
-                            <Stack space={32}>
-                                {breadcrumbs}
-                                {header}
-                            </Stack>
+                    ) : (
+                        <Stack space={header ? {mobile: 24, desktop: 32} : 0}>
+                            {mainContent}
                             {!isBleedActivated && extra}
                         </Stack>
-                    </Box>
-                )}
+                    )}
+                </Box>
             </ResponsiveLayout>
-            {isBleedActivated && extra && (isTabletOrSmaller || !sideBySideExtraOnDesktop) && (
+            {isBleedActivated && (
                 <ResponsiveLayout
+                    className={sideBySideExtraOnDesktop ? styles.hideOnDesktop : ''}
                     backgroundColor={`linear-gradient(to bottom, ${vars.colors.backgroundBrand} 40px, ${vars.colors.background} 0%)`}
                 >
                     {extra}
                 </ResponsiveLayout>
             )}
-        </>
+        </div>
     );
 };
 
 type MainSectionHeaderLayoutProps = {
     isInverse?: boolean;
     children: RendersElement<typeof MainSectionHeader>;
+    dataAttributes?: DataAttributes;
 };
 
 export const MainSectionHeaderLayout: React.FC<MainSectionHeaderLayoutProps> = ({
     isInverse = true,
     children,
+    dataAttributes,
 }) => {
-    const {isTabletOrSmaller} = useScreenSize();
-
     return (
-        <ResponsiveLayout isInverse={isInverse}>
+        <ResponsiveLayout
+            isInverse={isInverse}
+            dataAttributes={{'component-name': 'MainSectionHeaderLayout', ...dataAttributes}}
```

Comment:
Maybe you can use `getPrefixedDataAttributes` here?

Reply:
no, that would be wrong. That's already done internally in ResponsiveLayout